pandas/core/base.py

```diff
@@ -684,8 +684,14 @@ def _gotitem(self, key, ndim, subset=None):
         # with the same groupby
         kwargs = dict([(attr, getattr(self, attr))
                        for attr in self._attributes])
+
+        if self._groupby.ndim == 1 and self._groupby.obj.name == key:
```

Comment:
I hate checks like this :>

does
``groupby = self._groupby[key]`` not just work here?